### PR TITLE
Milestone0

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: implementation/common/src/test/java/org/inaetics/isep/D3BaseVectorTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package org.inaetics.isep;
```
Comment (jjkester) [inline implementation/common/src/test/java/org/inaetics/isep/D3BaseVectorTest.java]: Fix wrong test package

File: implementation/physicsengine/pom.xml [hunk 2]
```diff
@@ -25,27 +26,22 @@
...
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Export-Package>org.inaetics.dronessimulator.physicsengine,
+                                        org.inaetics.dronessimulator.common,
+                                        org.inaetics.dronessimulator.common.protocol,
+                                        org.inaetics.dronessimulator.pubsub.api,
+                                        org.inaetics.dronessimulator.pubsub.api.publisher,
+                                        org.inaetics.dronessimulator.pubsub.api.subscriber,
+                                        org.inaetics.dronessimulator.pubsub.api.serializer</Export-Package>
```
Comment (jjkester) [inline implementation/physicsengine/pom.xml]: Serializer package is not needed - only required by publisher and subscriber (Felix dependency manager)

Comment (jjkester) [reply implementation/physicsengine/pom.xml]: Test whether api.* works

File: implementation/physicsengine/src/java/org/inaetics/dronessimulator/physicsengine/Entity.java [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
 @AllArgsConstructor
 @Getter
 @Setter
 @ToString
-public class Entity {
+public class Entity implements Cloneable {
     // X == Width, y == depth, z == height
     private final int id;
     private final Size size;
+    private final boolean collideable;
```
Comment (jjkester) [inline implementation/physicsengine/src/java/org/inaetics/dronessimulator/physicsengine/Entity.java]: Typo? "collidable"

File: implementation/pubsub/rabbitmq/subscriber/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/subscriber/RabbitMessageConsumer.java [hunk 4]
```diff
@@ -11,15 +11,15 @@
 /**
  * A RabbitMQ consumer to work in conjunction with the RabbitMQ subscriber.
  */
-class RabbitMessageConsumer extends DefaultConsumer implements Runnable {
+public class RabbitMessageConsumer extends DefaultConsumer implements Runnable {
```
Comment (jjkester) [inline implementation/pubsub/rabbitmq/subscriber/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/subscriber/RabbitMessageConsumer.java]: Does not need to be public!

File: implementation/pubsub/rabbitmq/subscriber/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/subscriber/RabbitMessageConsumer.java [hunk 5]
```diff
@@ -11,15 +11,15 @@
...
     /** A RabbitMQ subscriber instance. */
     private RabbitSubscriber subscriber;
 
     /**
      * Instantiates a new RabbitMQ consumer based on the given subscriber.
      * @param subscriber The subscriber related to this consumer.
      */
-    RabbitMessageConsumer(RabbitSubscriber subscriber) {
+    public RabbitMessageConsumer(RabbitSubscriber subscriber) {
```
Comment (jjkester) [inline implementation/pubsub/rabbitmq/subscriber/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/subscriber/RabbitMessageConsumer.java]: Does not need to be public!

Comment (jjkester) [reply implementation/pubsub/rabbitmq/subscriber/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/subscriber/RabbitMessageConsumer.java]: Build RabbitSubscriber, `.connect()` should start stuff, `.disconnect()` should stop it. Topics are automatically registered. Handlers can be added to the RabbitSubscriber, used by the RabbitMessageConsumer.

File: implementation/visualisation/src/main/java/org/inaetics/dronessimulator/visualisation/Input.java [hunk 6]
```diff
@@ -134,7 +140,7 @@ public boolean isFireSecondaryWeapon() {
         return keyboardBitSet.get(secondaryWeaponKey.ordinal());
     }
 
-    private void processInput() {
+    public void processInput() {
```
Comment (jjkester) [inline implementation/visualisation/src/main/java/org/inaetics/dronessimulator/visualisation/Input.java]: Check whether this is intentionally private.

**Review by jjkester — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Om te packagen: bundles/bundle.sh
Om PhysicsEngine te runnen: verplaats bundles/{java-serializer-0.1.jar, rabbitmq-publisher-0.1.jar, physicsengine-0.1.jar} naar felix/bundle. Start daarna felix
Om visualisation te runnen: java -jar bundles/visualisation-0.1.jar

- Issue: RabbitMQ lijkt teveel geheugen & CPU te gebruiken bij maar 1 berichtje per 3 sec.
- RabbitMessageConsumer spammed basicConsume. Dit kan gezien worden door de huidige visualisatie te starten en in stdout te kijken
- handleDelivery, sendMessage and ACK'en lijken niet gespammed te worden

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java [hunk 7]
```diff
@@ -58,14 +58,19 @@ protected RabbitConnection(ConnectionFactory connectionFactory) {
      * @throws IOException Error while setting up the connection.
      */
     public void connect() throws IOException {
+        System.out.println("CONNECTING");
```
Comment (jjkester) [inline implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java]: Shouldn't the prints in this file be replaced with log calls?

File: implementation/pubsub/rabbitmq/publisher/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/publisher/RabbitPublisher.java [hunk 8]
```diff
@@ -52,6 +52,7 @@ public void send(Topic topic, Message message) {
             if (message != null && serializer != null) {
                 byte[] serializedMessage = serializer.serialize(message);
                 this.channel.basicPublish(topic.getName(), "", null, serializedMessage);
+                System.out.println("[Publisher] Send message");
```
Comment (jjkester) [inline implementation/pubsub/rabbitmq/publisher/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/publisher/RabbitPublisher.java]: Should be replaced with log calls

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java [hunk 9]
```diff
@@ -63,6 +63,7 @@ public void connect() throws IOException {
             try {
                 connection = connectionFactory.newConnection();
             } catch (ConnectException | TimeoutException e) {
+                e.printStackTrace();
```
Comment (jjkester) [inline implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java:66]: Should be replaced with log call

File: implementation/pubsub/rabbitmq/publisher/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/publisher/RabbitPublisher.java [hunk 10]
```diff
@@ -54,6 +54,7 @@ public void send(Topic topic, Message message) {
                 this.channel.basicPublish(topic.getName(), "", null, serializedMessage);
             }
         } catch (IOException ignore) {
+            ignore.printStackTrace();
```
Comment (jjkester) [inline implementation/pubsub/rabbitmq/publisher/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/publisher/RabbitPublisher.java:57]: Should be replaced with log call

**Review by jjkester — APPROVED** [review]
Have some suggestions but can be merged

**Pull request merged**